src/UI/App.jsx

```diff
@@ -304,14 +311,26 @@ export default class App extends React.Component {
 		} else if(path[layer0] === "xIrisIdentification") {
 			objectToSaveIntoState = {deviceIdentification: value};
 		} else if(path[layer0] === "userAdditionalInformation") {
-			objectToSaveIntoState = {userAdditionalInformation: {}};
-			objectToSaveIntoState.userAdditionalInformation
+			const userAdditionalInformation
 				= JSON.parse(JSON.stringify(window.parleySettings.userAdditionalInformation));
+			objectToSaveIntoState = {userAdditionalInformation};
 
-			// For `userAdditionalInformation` we don't care about validating
-			// the contents. So we can just directly add it to the state.
 			// We're using JSON.parse(JSON.stringify()) to remove the Proxy
 			// from the object
+
+			// TODO: When these are `deepMerged` below, it means you can't remove
+			//  stuff from the setting object..
+			//  merge {} with {"oldProperty": "oldValue"} does not end up as {} in the state ...
+		} else if(path[layer0] === "apiCustomHeaders") {
+			const apiCustomHeaders = JSON.parse(JSON.stringify(window.parleySettings.apiCustomHeaders));
+			objectToSaveIntoState = {apiCustomHeaders};
+
+			// We're using JSON.parse(JSON.stringify()) to remove the Proxy
+			// from the object
+
+			// TODO: When these are `deepMerged` below, it means you can't remove
```

Comment:
@daanleenders wil je nog even naar deze todo kijken en of we dit erg vinden? Anders laten we het staan (staat nu namelijk in de readme uitgelegd)

Reply:
Ja in readme/documentatie is prima. Dezen kunnen idd weg.

Reply:
Done